### ISSUE TITLE
Crash when sharing without link parameter (API defines link parameter as optional)

### ISSUE THREAD

**Issue by VilemKurz** [pinterest/ios-pdk#47]
PDKClient defines a method (e.g, unfortunately this is the case with all sharing methods)
```objectivec
- (void)createPinWithImage:(UIImage *)image
                      link:(NSURL *)link
                   onBoard:(NSString *)boardId
               description:(NSString *)pinDescription
                  progress:(PDKPinUploadProgress)progressBlock
               withSuccess:(PDKClientSuccess)successBlock
                andFailure:(PDKClientFailure)failureBlock;
```
When you supply UIImage, but not link (in [API doc](https://developers.pinterest.com/docs/api/pins/) marked *optional*), the sdk crashes.

**Comment by rcancro** [comment]
Good catch. Looks like we are inserting link in a dictionary without checking for nil. Would you mind submitting a pull request?

**Comment by connor** [comment]
Fixed in #57.
